### PR TITLE
FEATURE/Add: experimental boomerang (BCT) distinguisher checker

### PR DESCRIPTION
**Pull request by juaninf**
## What

Adds `boomerang_checker_for_block_cipher_single_key` to `claasp/cipher_modules/models/utils.py`, completing the family of single-key experimental distinguisher checkers (which already covers differential-linear and linear).

## How

It mirrors the existing checkers and reuses their helpers (`_repeat_input_difference`, `_parallel_dispatch`). For a fixed key it draws random plaintext pairs `P1 = P0 ^ Δ`, encrypts with `evaluate_vectorized`, applies the output difference `∇` to both ciphertexts, decrypts with `cipher.cipher_inverse()`, and counts the fraction of quartets that return exactly by `Δ`:

```
P1 = P0 ^ input_difference
C0, C1 = E(P0), E(P1)
P2, P3 = E⁻¹(C0 ^ output_difference), E⁻¹(C1 ^ output_difference)
success  ⇔  (P2 ^ P3) == input_difference
```

Returns the empirical boomerang probability in `[0, 1]`. The only capability it needs beyond the differential-linear checker is `cipher.cipher_inverse()` (already available), so it is **generic for any invertible block cipher / permutation**. Parallel workers via the shared `_parallel_dispatch`.

## Tests

`tests/unit/cipher_modules/models/models_utils_test.py::test_boomerang_checker_for_block_cipher_single_key` — on **Speck-32/64**:
- deterministic edge cases (a zero difference on either side returns `1.0`), including the parallel `num_workers=2` path;
- a **rounds-4→7 sub-distinguisher** (through the switch) of the SPECK32/64 boomerang characteristic from Wang, Wang, Sun, *"SAT-aided Automatic Search of Boomerang Distinguishers for ARX Ciphers"* (ToSC), **Table 6** — round-4 diff `0x81008102`, round-7 diff `0x0a040804`, boomerang probability ≈ `2^-14`; single-key, so a 3-round reduced Speck reproduces it;
- input validation (`ValueError` on non-multiple-of-8 block/key sizes).

Doctest (`sage -t`) and the unit test pass in the CLAASP docker image.

### CODE REVIEW

File: claasp/cipher_modules/models/utils.py [hunk 1]
```diff
@@ -1122,6 +1122,85 @@ def linear_checker_for_block_cipher_single_key(
...
+    OUTPUT:
+
+    - This method returns a **float**; the empirical boomerang probability in the interval ``[0, 1]``
+
+    EXAMPLES::
+
+        sage: from claasp.ciphers.block_ciphers.speck_block_cipher import SpeckBlockCipher
+        sage: from claasp.cipher_modules.models.utils import boomerang_checker_for_block_cipher_single_key
+        sage: speck = SpeckBlockCipher(number_of_rounds=5)
+        sage: boomerang_checker_for_block_cipher_single_key(speck, 0, 0, 1024, 32, 64, 0)
```
Comment (peacker) [inline claasp/cipher_modules/models/utils.py]: make the doctest self-explanatory, e.g.:
        sage: input_difference = 0
        sage: boomerang_checker_for_block_cipher_single_key(speck, 0, input_difference, 1024, 32, 64, 0)
add names to all numerical parameters in the function call

**Review by peacker — CHANGES_REQUESTED** [review]
address sonar comment and the comment on the doctest

**Review by peacker — APPROVED** [review]
